core/src/main/java/org/mule/runtime/core/internal/context/thread/notification/ThreadsStatistics.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.internal.context.thread.notification;
+
+import org.apache.commons.math3.stat.StatUtils;
+import org.mule.runtime.api.util.Pair;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static java.lang.Math.sqrt;
+
+public class ThreadsStatistics {
+
+  private Map<Pair<String, String>, List<Long>> times = new HashMap<>();
```

Comment:
is there a way to clear this data? foir instance, to start getting metrics only after warm-up of the vm is done